com.avaloq.tools.ddk.xtext.export/src/com/avaloq/tools/ddk/xtext/export/generator/ResourceDescriptionStrategyGenerator.xtend

```diff
@@ -83,7 +83,8 @@ class ResourceDescriptionStrategyGenerator {
                 @Override
                 public Boolean case«c.type.name»(final «c.type.instanceClassName()» obj) {
                   «val guard = c.guard.javaExpression(ctx.clone('obj', c.type))»
-                  «IF c.guard !== null && !guard.equalsIgnoreCase("false")»
+                  «IF guard.equalsIgnoreCase("false")»
+                  «ELSEIF c.guard !== null»

```

Comment:
Having an empty conditional block looks wrong to me. How about replacing the «ELSE» on line 91 with «ELSEIF c.guard == null» ?